src/code/z_bgcheck.c

```diff
@@ -929,9 +928,8 @@ s32 BgCheck_CheckLineAgainstSSList(SSList* ssList, CollisionContext* colCtx, u16
     curNode = &colCtx->polyNodes.tbl[ssList->head];
     while (true) {
         polyId = curNode->polyId;
-        checkedPoly = &colCtx->polyNodes.polyCheckTbl[polyId];
 
-        if (*checkedPoly == true || COLPOLY_VIA_FLAG_TEST(polyList[polyId].flags_vIA, xpFlags1) ||
+        if (COLPOLY_VIA_FLAG_TEST(polyList[polyId].flags_vIA, xpFlags1) ||
             !(xpFlags2 == 0 || COLPOLY_VIA_FLAG_TEST(polyList[polyId].flags_vIA, xpFlags2))) {
 
             if (curNode->next == SS_NULL) {
```

Comment:
Not one of your changes but this if statement looks better like that imo:
```c
if (curNode->next != SS_NULL) {
    curNode = &colCtx->polyNodes.tbl[curNode->next];
} else {
    break;
}
```
it shouldn't change anything afaik

I don't have anything else to say, really cool to have performance improvements 😄

Reply:
just in case it's not clear I'm speaking of this if:
```c
if (curNode->next == SS_NULL) {
    break;
} else {
    curNode = &colCtx->polyNodes.tbl[curNode->next];
    continue;
}
```